### PR TITLE
Framework migration of wazuh-db module

### PR DESCRIPTION
**Pull request by lsayanes**
This PR includes all the necessary changes to give support to the `wazuh-db` migrated tests in the wazuh/wazuh repository. Those tests will use this new structure.

### CODE REVIEW

File: src/wazuh_testing/constants/executions.py
Comment (TomasTurina) [inline src/wazuh_testing/constants/executions.py:1]: I think this file shouldn't be part of the framework.

File: src/wazuh_testing/tools/api.py
Comment (TomasTurina) [inline src/wazuh_testing/tools/api.py:1]: Remove this file. Use functions from `modules/api/utils.py`. Check comments from https://github.com/wazuh/qa-integration-framework/pull/48 that also apply here.

File: src/wazuh_testing/utils/agent_groups.py [hunk 1]
```diff
@@ -0,0 +1,30 @@
+import subprocess
```
Comment (TomasTurina) [inline src/wazuh_testing/utils/agent_groups.py:1]: Missing header.

File: src/wazuh_testing/utils/agent_groups.py
Comment (TomasTurina) [inline src/wazuh_testing/utils/agent_groups.py:1]: Rebase branch. This file already exists and is not necessary to create it again.

File: src/wazuh_testing/utils/configuration.py [hunk 2]
```diff
@@ -409,3 +411,55 @@ def update_feed_path_configurations(configurations, metadata, feeds_path):
                                                    os.path.join(feeds_path, 'nvd', metadata[index]['nvd_feed'])))
 
     return new_configurations
+
+
+def set_correct_prefix(configurations, new_prefix):
```
Comment (TomasTurina) [inline src/wazuh_testing/utils/configuration.py:416]: Remove empty function.

File: src/wazuh_testing/utils/configuration.py [hunk 3]
```diff
@@ -409,3 +411,55 @@ def update_feed_path_configurations(configurations, metadata, feeds_path):
...
+    Args:
+        configurations (list): List of configurations loaded from the YAML.
+        new_prefix (str): Prefix to be inserted before every path.
+
+    Returns:
+        configurations (list): List of configurations with the correct prefix added in the directories and
+        ignore sections.
+    """
+
+def load_wazuh_configurations(yaml_file_path: str, test_name: str, params: list = None, metadata: list = None) -> object():
```
Comment (TomasTurina) [inline src/wazuh_testing/utils/configuration.py:435]: Remove this function. Use `load_configuration_template` instead or adapt it if necessary.

File: src/wazuh_testing/utils/database.py [hunk 4]
```diff
@@ -123,3 +123,12 @@ def run_sql_script(database_path: Union[os.PathLike, str], script_path: Union[os
     """
     with DatabaseAdministrator(database_path) as db:
         db.execute_script(script_path)
+
+def validate_interval_format(interval):
```
Comment (TomasTurina) [inline src/wazuh_testing/utils/database.py:127]: This function should be in `time.py` file. Remove `import numbers` once moved there.

File: src/wazuh_testing/utils/db_queries/global_db.py [hunk 5]
```diff
@@ -69,3 +70,100 @@ def delete_agent(agent_id):
         agent_id (str): Agent ID.
     """
     database.query_wdb(f"global sql DELETE FROM agent where id={int(agent_id)}")
+
+
+def clean_agents_from_db():
```
Comment (TomasTurina) [inline src/wazuh_testing/utils/db_queries/global_db.py:75]: Rebase branch. This function already exists.

File: src/wazuh_testing/utils/db_queries/global_db.py [hunk 6]
```diff
@@ -69,3 +70,100 @@ def delete_agent(agent_id):
...
+    """
+    command = 'global sql DELETE FROM agent WHERE id != 0'
+    try:
+        database.query_wdb(command)
+    except Exception:
+        raise Exception('Unable to clean agents')
+
+
+
+def insert_agent_in_db(id=1, name='TestAgent', ip='any', registration_time=0, connection_status=0,
```
Comment (TomasTurina) [inline src/wazuh_testing/utils/db_queries/global_db.py:87]: Remove. Use `create_or_update_agent`.

File: src/wazuh_testing/utils/db_queries/global_db.py [hunk 7]
```diff
@@ -69,3 +70,100 @@ def delete_agent(agent_id):
...
+                       disconnection_time = "{disconnection_time}" WHERE id = {id};'
+    try:
+        database.query_wdb(insert_command)
+        database.query_wdb(update_command)
+    except Exception:
+        raise Exception(f"Unable to add agent {id}")
+
+
+# Insert agents into DB and assign them into a group
+def insert_agent_into_group(total_agents):
```
Comment (TomasTurina) [inline src/wazuh_testing/utils/db_queries/global_db.py:103]: Create a function to only set agent groups (for creating agents there is `create_or_update_agent`). Parametrize everything, this function should be generic.

File: src/wazuh_testing/utils/db_queries/global_db.py [hunk 8]
```diff
@@ -69,3 +70,100 @@ def delete_agent(agent_id):
...
+        results = database.query_wdb(command)
+        assert results == 'ok'
+
+        command = f'''global set-agent-groups {{"mode":"append","sync_status":"syncreq",
+                   "source":"remote","data":[{{"id":{id},"groups":["Test_group{id}"]}}]}}'''
+        results = database.query_wdb(command)
+        assert results == 'ok'
+
+
+def remove_db_agent(agent_id):
```
Comment (TomasTurina) [inline src/wazuh_testing/utils/db_queries/global_db.py:118]: Remove. Use `delete_agent` instead (after rebase).

File: src/wazuh_testing/utils/file.py [hunk 9]
```diff
@@ -465,3 +465,36 @@ def copy_files_in_folder(src_folder, dst_folder='/tmp', files_to_move=None):
...
+
+    Returns:
+       list: Yaml structure.
+    """
+    value_list = []
+    with open(file_path) as f:
+        value_list.append((yaml.safe_load(f), file_path.split(separator)[0]))
+
+    return value_list
+
```
Comment (TomasTurina) [inline src/wazuh_testing/utils/file.py:486]: Missing empty space.

File: src/wazuh_testing/utils/file.py [hunk 10]
```diff
@@ -465,3 +465,36 @@ def copy_files_in_folder(src_folder, dst_folder='/tmp', files_to_move=None):
...
+def recursive_directory_creation(path):
```
Comment (TomasTurina) [inline src/wazuh_testing/utils/file.py:487]: Rebase and remove. This function already exists.

File: src/wazuh_testing/utils/manage_agents.py [hunk 11]
```diff
@@ -47,3 +47,7 @@ def remove_agents(agents_id: List, remove_type: str = 'wazuhdb') -> None:
             response_data = response.json()
             if response.status_code != 200:
                 raise RuntimeError(f"Error deleting an agent: {response_data}")
+
+
+def remove_all_agents(remove_type):
```
Comment (TomasTurina) [inline src/wazuh_testing/utils/manage_agents.py:52]: Remove. Use `clean_agents_from_db` instead (after rebase).

File: src/wazuh_testing/utils/db_queries/global_db.py [hunk 12]
```diff
@@ -69,3 +70,100 @@ def delete_agent(agent_id):
...
+    Clean belong table from global.db
+    """
+    command = 'global sql DELETE FROM belongs'
+    try:
+        database.query_wdb(command)
+    except Exception:
+        raise Exception('Unable to clean belongs table.')
+
+
+def list_agents_ids():
```
Comment (TomasTurina) [inline src/wazuh_testing/utils/db_queries/global_db.py:165]: Remove. Not necessary if you use `clean_agents_from_db` (after rebase).

File: src/wazuh_testing/utils/services.py [hunk 13]
```diff
@@ -14,6 +14,7 @@
 from wazuh_testing.constants.paths.sockets import WAZUH_SOCKETS, WAZUH_OPTIONAL_SOCKETS
 from wazuh_testing.constants.paths.variables import VAR_RUN_PATH, VERSION_FILE
 from wazuh_testing.constants.platforms import MACOS, SOLARIS, WINDOWS
+from wazuh_testing.constants.paths import WAZUH_PATH
```
Comment (TomasTurina) [inline src/wazuh_testing/utils/services.py:17]: Why is this?

File: src/wazuh_testing/utils/time.py [hunk 14]
```diff
@@ -2,33 +2,237 @@
...
+import subprocess
+import sys
+import threading
+import time
+from datetime import datetime, timedelta
 
 
-def parse_date_time_format(date_time: str):
-    """Parse the specified date_time to return a common format.
+class TimeMachine:
```
Comment (TomasTurina) [inline src/wazuh_testing/utils/time.py:14]: There is this open issue: https://github.com/wazuh/wazuh-qa/issues/4160

Can we avoid using this? Maybe another solution?

**Review by TomasTurina — CHANGES_REQUESTED** [review]
Rebase branch.

File: src/wazuh_testing/utils/configuration.py [hunk 15]
```diff
@@ -5,11 +5,13 @@
 import sys
 import json
 import xml.etree.ElementTree as ET
+import yaml
```
Comment (TomasTurina) [inline src/wazuh_testing/utils/configuration.py:8]: Remove import.

File: src/wazuh_testing/utils/configuration.py [hunk 16]
```diff
@@ -5,11 +5,13 @@
...
 
 from copy import deepcopy
 from typing import List
 
 from wazuh_testing import DATA_PATH
+from wazuh_testing.constants.paths import ROOT_PREFIX
```
Comment (TomasTurina) [inline src/wazuh_testing/utils/configuration.py:14]: Remove import.

File: src/wazuh_testing/utils/database.py [hunk 17]
```diff
@@ -7,7 +7,6 @@
 import json
 import socket
 import time
-
```
Comment (TomasTurina) [inline src/wazuh_testing/utils/database.py:10]: Remove this change.

File: src/wazuh_testing/utils/time.py [hunk 18]
```diff
@@ -53,3 +258,51 @@ def get_current_timestamp() -> int:
...
+    ]
+
+    for item in regex_list:
+        match = re.compile(item['regex']).match(date_time)
+
+        if match:
+            return f"{match.group(1)} {match.group(2)}{item['append']}"
+
+    ValueError(f"Could not parse the {date_time} datetime.")
+
```
Comment (TomasTurina) [inline src/wazuh_testing/utils/time.py:300]: Missing empty space.

File: src/wazuh_testing/utils/database.py (hunk 17 above)
Comment (mateocervilla) [reply src/wazuh_testing/utils/database.py:10]: What do you mean? I removed an unused import `import numbers`

**Review by TomasTurina — APPROVED** [review]

File: src/wazuh_testing/utils/db_queries/agent_db.py [hunk 19]
```diff
@@ -290,3 +291,30 @@ def rootcheck_delete(agent_id: str = '000'):
         agent_id (str): Agent ID.
     """
     database.query_wdb(f"agent {agent_id} rootcheck delete")
+
+
+def agent_checksum_data(agent_id: str = '001', path: str = '/home/test/file'):
```
Comment (TomasTurina) [inline src/wazuh_testing/utils/db_queries/agent_db.py:296]: This function is not generic, please modify it.

**Review by TomasTurina — CHANGES_REQUESTED** [review]

**Comment by mauromalara** [comment]
@mateocervilla @TomasTurina Please remove my name as a reviewer as I am no longer affiliated with Wazuh 🙏🏻 

Thanks!

**Comment by TomasTurina** [comment]
Please, change base branch to 4.9.0.

**Pull request merged**